File: src/main/java/dk/itealisten/xmlnormalize/Node.java
package dk.itealisten.xmlnormalize;

import java.util.ArrayList;
import java.util.List;

public class Node {


    private final String name;

    public Node(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    private List<Node> children = new ArrayList<>();

    public Node addChild(Node child) {
        children.add(child);
        return this;
    }

    public List<Node> getChildren() {
        return children;
    }

    public String toString() {
        return "Node(name='" + name + "', children=" + children.toString() + ")";
    }
}

File: src/main/kotlin/dk/itealisten/xmlnormalize/Node.kt
package dk.itealisten.xmlnormalize

class Node
constructor(val name: String) {

    private val children = ArrayList<Node>()

    fun addChild(child: Node): Node {
        children.add(child)
        return this
    }

    fun getChildren(): Array<Node> {
        return children.toTypedArray()
    }

    override fun toString(): String {
        return "Node(name='$name', children=$children)"
    }


}
